homeassistant/components/waqi/config_flow.py

```diff
@@ -0,0 +1,106 @@
+from homeassistant.helpers.selector import LocationSelector
+from homeassistant.helpers.typing import ConfigType
+
+from .const import CONF_STATION_NUMBER, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class WAQIConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for World Air Quality Index (WAQI)."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            async with WAQIClient(
+                session=async_get_clientsession(self.hass)
+            ) as waqi_client:
+                waqi_client.authenticate(user_input[CONF_API_KEY])
+                try:
+                    location = user_input[CONF_LOCATION]
+                    measuring_station: WAQIAirQuality = (
+                        await waqi_client.get_by_coordinates(
+                            location[CONF_LATITUDE], location[CONF_LONGITUDE]
+                        )
+                    )
+                except WAQIAuthenticationError:
+                    errors["base"] = "invalid_auth"
+                except WAQIConnectionError:
+                    errors["base"] = "cannot_connect"
+                except Exception as exc:  # pylint: disable=broad-except
+                    _LOGGER.exception(exc)
+                    errors["base"] = "unknown"
+                else:
+                    await self.async_set_unique_id(str(measuring_station.station_id))
+                    self._abort_if_unique_id_configured()
+                    return self.async_create_entry(
+                        title=measuring_station.city.name,
+                        data={
+                            CONF_API_KEY: user_input[CONF_API_KEY],
+                            CONF_STATION_NUMBER: measuring_station.station_id,
+                        },
+                    )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=self.add_suggested_values_to_schema(
+                vol.Schema(
+                    {
+                        vol.Required(CONF_API_KEY): str,
+                        vol.Required(
+                            CONF_LOCATION,
+                        ): LocationSelector(),
+                    }
+                ),
+                user_input
+                or {
+                    CONF_LOCATION: {
+                        CONF_LATITUDE: self.hass.config.latitude,
+                        CONF_LONGITUDE: self.hass.config.longitude,
+                    }
+                },
+            ),
+            errors=errors,
+        )
+
+    async def async_step_import(self, import_config: ConfigType) -> FlowResult:
+        """Handle importing from yaml."""
+        await self.async_set_unique_id(str(import_config[CONF_STATION_NUMBER]))
+        self._abort_if_unique_id_configured()
```

Comment:
If I create 1 issue for every measuring station that already exists, users can get 4 issues, is that what we want?

Reply:
I think not but if you use the same issue id it will just update the existing one and don't create four off them.

Reply:
Ah good one, I can add an issue then